### PR TITLE
Pages for amending deceased settlor

### PR DESCRIPTION
**Pull request by adamconder**

### CODE REVIEW

File: app/forms/DateOfDeathFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  def withConfig(trustStartDate: LocalDate, prefix: String): Form[LocalDate] =
+    Form(
+      "value" -> localDate(
+        invalidKey     = s"$prefix.error.invalid",
+        allRequiredKey = s"$prefix.error.required.all",
+        twoRequiredKey = s"$prefix.error.required.two",
+        requiredKey    = s"$prefix.error.required"
+      ).verifying(firstError(
+        maxDate(trustStartDate, s"$prefix.error.future", "day", "month", "year"),
+        minDate(LocalDate.of(1500,1,1), s"$prefix.error.past", "day", "month", "year")
```
Comment (jackheslop96) [inline app/forms/DateOfDeathFormProvider.scala:36]: Should the min date not be their date of birth? Or are we not guaranteed to have that?

File: app/navigation/DeceasedSettlorNavigator.scala [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    routes(page)(userAnswers)
+  }
+
+  private val simpleNavigation: PartialFunction[Page, Call] = {
+    case NamePage => ???
+  }
+
+  private val yesNoNavigation: PartialFunction[Page, UserAnswers => Call] = {
+    case _ => ???
+  }
```
Comment (jackheslop96) [inline app/navigation/DeceasedSettlorNavigator.scala:43]: Still to do?

File: app/pages/individual/deceased/DateOfDeathYesNoPage.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+case object DateOfDeathYesNoPage extends QuestionPage[Boolean] {
+
+  override def path: JsPath = basePath \ toString
+
+  override def toString: String = "dateOfDeathYesNo"
+
```
Comment (jackheslop96) [inline app/pages/individual/deceased/DateOfDeathYesNoPage.scala:27]: Missing cleanup of DateOfDeathPage?

File: conf/messages.en [hunk 4]
```diff
@@ -352,4 +352,89 @@ businessSettlor.checkDetails.heading = Check business settlor details
 
 removeBusinessSettlor.title = Are you sure you want to remove the business settlor?
 removeBusinessSettlor.heading = Are you sure you want to remove {0}?
-removeBusinessSettlor.error.required = Select yes if you want to remove the business settlor
\ No newline at end of file
+removeBusinessSettlor.error.required = Select yes if you want to remove the business settlor
+
+
+# Adam todo come back to this what should the default be?
+deceasedSettlor.name.default = the individual settlor
```
Comment (jackheslop96) [inline conf/messages.en:359]: To do

File: app/forms/DateOfDeathFormProvider.scala (hunk 1 above)
Comment (adamconder) [reply app/forms/DateOfDeathFormProvider.scala:36]: Good question I'll ask Wendy

File: app/navigation/DeceasedSettlorNavigator.scala (hunk 2 above)
Comment (adamconder) [reply app/navigation/DeceasedSettlorNavigator.scala:43]: Yes :)

File: app/pages/individual/deceased/DateOfDeathYesNoPage.scala (hunk 3 above)
Comment (adamconder) [reply app/pages/individual/deceased/DateOfDeathYesNoPage.scala:27]: Doing it now

File: conf/messages.en (hunk 4 above)
Comment (adamconder) [reply conf/messages.en:359]: Need to speak with Marianne

**Pull request merged**